src/Oracle/Oracle.cs

```diff
@@ -0,0 +1,377 @@
+using Akka.Actor;
+using Akka.Util.Internal;
+using Microsoft.Extensions.Configuration;
+using Neo.ConsoleService;
+using Neo.Cryptography;
+using Neo.Cryptography.ECC;
+using Neo.IO;
+using Neo.IO.Json;
+using Neo.Ledger;
+using Neo.Network.P2P;
+using Neo.Network.P2P.Payloads;
+using Neo.Persistence;
+using Neo.SmartContract;
+using Neo.SmartContract.Native;
+using Neo.SmartContract.Native.Oracle;
+using Neo.VM;
+using Neo.Wallets;
+using Neo.Wallets.NEP6;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Timers;
+using static System.IO.Path;
+
+namespace Neo.Plugins
+{
+    public class Oracle : Plugin, IPersistencePlugin
+    {
+        private NEP6Wallet Wallet;
+        private string[] Nodes;
+        private TimeSpan MaxTaskTimeout;
+        private ConcurrentDictionary<ulong, OracleTask> PendingQueue;
+        private CancellationTokenSource CancelSource;
+        private int Counter;
+
+        private static readonly OracleHttpProtocol Https = new OracleHttpProtocol();
+
+        public override string Description => "Built-in oracle plugin";
+
+        public Oracle()
+        {
+            PendingQueue = new ConcurrentDictionary<ulong, OracleTask>();
+
+            RpcServerPlugin.RegisterMethods(this);
+
+            System.Timers.Timer timer = new System.Timers.Timer();
+            timer.Enabled = true;
+            timer.Interval = 1000 * 60 * 3;
+            timer.Start();
+            timer.Elapsed += new ElapsedEventHandler(OnTimer);
+        }
+
+        protected override void Configure()
+        {
+            var config = GetConfiguration();
+            Wallet = new NEP6Wallet(Combine(PluginsDirectory, nameof(Oracle), config.GetSection("Wallet").Value));
+            Nodes = config.GetSection("Nodes").GetChildren().Select(p => p.Get<string>()).ToArray();
```

Comment:
I don't think this is a viable long-term solution. This module desperately needs neo-project/neo#1573. We plan to start working on it soon (as NeoFS needs this too).

Reply:
Look forward to hearing that.